### DISCUSSION TITLE
Is it really necessary to update the kind 30000 offer's status as the trade progresses?

### DISCUSSION THREAD

**Discussion by nobu-maeda** [MostroP2P/mostro#53] [Ideas]
As titled. Perhaps it would improve privacy to not do so. And perhaps an offer can even be wiped off the servers once taken? Starting a discussion instead of an issue for this.

**Comment by grunch** [comment]
the main reason is we need to remove the Pending order after a user took it, it would be great that both parties knows the status of the order at anytime while the order is progressing.

If we want to add more privacy the current flow could be replaced by:
* Order status: Pending
* Buyer take the order
* Mostro replaces the order with a new status `NotPending` or whatever, we just want not give information to a third user (the order could expired, taken or canceled), mostro internally knows the status of the order, and the order will not appear anymore on the order book
* Both parties can send to Mostro a Message with an Action via DM asking the current status, also Mostro can respond any Message with the current status of the order.

**Reply by nobu-maeda** [reply]
Can't the status be known between the buyer and seller just by them DMing each other? For the purpose of everybody else, the order is already taken. If the order falls through, the maker can always repost?

**Reply by grunch** [reply]
why would you trust your counterpart? What I think now is that users can ask for the order status asking to the Mostro in a DM

And yes, if the order falls through it can be reposted

**Reply by nobu-maeda** [reply]
> why would you trust your counterpart? What I think now is that users can ask for the order status asking to the Mostro in a DM
> 
> And yes, if the order falls through it can be reposted

Definitely not trusting the counterparty. Each trading party should know the status and where in the state machine for itself. I am just wondering why publishing an update to relays globally regarding the status is necessary. Once an offer is taken and trade starts, any additional information published to relays is just extra information to doxx trading parties out of?

**Reply by grunch** [reply]
I agree, thanks for this suggestion, I Will change that in code

**Comment by grunch** [comment]
I proposed this NIP where we only maintain 4 statuses

https://mostro.network/protocol/order_event.html
